pkgs/development/tools/marksman/default.nix

```diff
@@ -0,0 +1,59 @@
+{ lib
+, fetchFromGitHub
+, buildDotnetModule
+, dotnetCorePackages
+, marksman
+, testVersion
+}:
+
+buildDotnetModule rec {
+  pname = "marksman";
+  version = "2022-10-30";
```

Comment:
```suggestion
  version = "unstable-2022-10-30";
```

Reply:
I didn't manage to build it with this version.

1. Version starting with a string is not allowed in dotnet.
1. Looks like dotnet is using `Version` or `version` env to lookup a version, or a tag from project file. But in a build phase nix overrides even that:
https://github.com/NixOS/nixpkgs/blob/b457130e8a21608675ddf12c7d85227b22a27112/pkgs/build-support/dotnet/build-dotnet-module/hooks/dotnet-build-hook.sh#L23-L25

And to be honest I don't like it much. I know about the rule:

> If a package is not a release but a commit from a repository, then the version attribute must be the date of that (fetched) commit. The date must be in `unstable-YYYY-MM-DD` format.

But it's not a random commit, it's a tagged release on github which happens to be a date. But yes, it looks like there is no traditional versioning for this project yet.

Any suggestions?

Reply:
`dotnetBuildFlags` seem to come after that version flag, does dotnet allow overriding arguments so you can hard-code your own version flag to override the one the build phase sets?

Reply:
I already tried that without luck. I tested it again and my problem was with configure phase. With `dotnetFlags` it works. But it somewhat complicates things, especially with update script.

Reply:
OK. Even if I don't like it, since no one is answering, I've added `unstable-` prefix. I hope it can be merged now.